cluster/plugin/local/plugin.go

```diff
@@ -150,7 +150,7 @@ func RunAgent(ctx context.Context, c *client.Client, action string, opts *Reques
 	reader, err := c.ImagePull(ctx, image, types.ImagePullOptions{})
 	if err != nil {
 		// don't exit, if not in the registry we may still want to run the container with a local image
-		fmt.Printf("ImagePull failed, which is fine if using a development version: %f", err)
+		fmt.Println("ampagent image pull failed, which is expected on a development version")
 	} else {
 		// wait for the image to be pulled
```

Comment:
You need defer reader.Close() here.

The make slice statement can be succinct since capacity defaults to length (`make([]byte, 1000)`). Related to this, In several other areas where you make channels, it's a bit more idiomatic to just use `make(chan <type>)` without explicitly specifying 1 for unbuffered channels.

But instead of doing this anyway, maybe import `github.com/docker/docker/pkg/jsonmessage` and then try using the following to print the response:

```
jsonmessage.DisplayJSONMessagesStream(
		reader,
		os.Stdout,
		os.Stdout.Fd(),
		false,
		nil)
```

Finally, even if we're going to refactor this or potentially eliminate it later, you can improve the code density / readability a bit by factoring out this image pull logic into a separate func.